examples/hackernews/back-end/reactive_service/src/hackernews.service.ts

```diff
@@ -26,38 +25,33 @@ type Upvote = {
 
 type Upvoted = Post & { upvotes: number; author: User };
 
-export default class HackerNewsService implements SkipService {
-  initialData: {
-    posts: Entry<string, Post>[];
-    users: Entry<string, User>[];
-    upvotes: Entry<string, Upvote>[];
-  };
-  resources = { posts: PostsResource };
-
-  constructor(
-    posts: Entry<string, Post>[],
-    users: Entry<string, User>[],
-    upvotes: Entry<string, Upvote>[],
-  ) {
-    this.initialData = { posts, users, upvotes };
-  }
-
-  createGraph(inputCollections: {
-    posts: EagerCollection<number, Post>;
-    users: EagerCollection<number, User>;
-    upvotes: EagerCollection<number, Upvote>;
-  }): Record<string, EagerCollection<Json, Json>> {
-    const upvotes = inputCollections.upvotes.map(UpvotesMapper);
-    const postsWithUpvotes = inputCollections.posts.map(
-      PostsMapper,
-      inputCollections.users,
-      upvotes,
-    );
+type Inputs = {
+  posts: EagerCollection<number, Post>;
+  users: EagerCollection<number, User>;
+  upvotes: EagerCollection<number, Upvote>;
+};
+type Outputs = {
+  postsWithUpvotes: EagerCollection<number, Upvoted>;
+};
```

Comment:
This is nice, but the name `Outputs` emphasizes the role as the result of `SkipService.createGraph` at the expense of the role as the argument of `Resource.instantiate`. It would be better to use a name that is good for both I think, and if anything, the resource role might be the more important one.

Maybe rename `Inputs` to `InitialInputs` and `Outputs` to `ResourceInputs`.